File: src/problems_all/217_contains_duplicate/217_contains_duplicate.go
/*
https://leetcode.com/problems/contains-duplicate/description/

Runtime: 105 ms
Memory: 9.6 MB
https://leetcode.com/problems/contains-duplicate/submissions/881242084/
*/

package main

import "fmt"

func main() {
	nums := []int{1, 2, 3, 4}
	result := containsDuplicate(nums)
	fmt.Println(result)
}

func containsDuplicate(nums []int) bool {
	counter := map[int]int8{}
	for _, num := range nums {
		if _, ok := counter[num]; ok {
			return true
		} else {
			counter[num] = 1
		}
	}

	return false
}

File: src/problems_all/217_contains_duplicate/217_contains_duplicate.py
"""
https://leetcode.com/problems/contains-duplicate/description/

Runtime: 584 ms
Memory: 26.1 MB
https://leetcode.com/problems/contains-duplicate/submissions/678186913/
"""

from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        num_set = set(nums)
        return len(nums) != len(num_set)


def main():
    nums = [1, 2, 3, 1]
    result = Solution().containsDuplicate(nums)
    print(result)


if __name__ == '__main__':
    main()
